visidata/features/dedupe.py

```diff
@@ -102,9 +102,21 @@ def _reload(self=vs):
     return vs
 
 
+@Sheet.api
+@asyncthread
+def hide_unvarying_cols(sheet):
+    if len(sheet.rows) < 2:
+        return
+    for col in sheet.visibleCols:
+        vals = (col.getTypedValue(r) for r in sheet.rows)
+        first = next(vals)
+        if all(v == first for v in vals):
+            col.hide()
+
 # Add longname-commands to VisiData to execute these methods
 BaseSheet.addCommand(None, "select-duplicate-rows", "sheet.select_duplicate_rows()", "select each row that is a duplicate of a prior row")
 BaseSheet.addCommand(None, "dedupe-rows", "vd.push(sheet.dedupe_rows())", "open new sheet in which only non-duplicate rows in the active sheet are included")
+BaseSheet.addCommand(None, 'hide-unvarying-cols', 'sheet.hide_unvarying_cols()', 'hide any column that has multiple rows but only one distinct value')
```

Comment:
These commands should all be on TableSheet instead of BaseSheet.

Reply:
Okay. If I move the new command to `features/layout.py` (see below), `Sheet` is used for `Sheet.addCommand()` there, so I'll use that to match. What's the intended distinction between `Sheet` and `TableSheet`?

Reply:
There's no functional difference between Sheet and TableSheet.  Sheet is the original name, in use in many places and not worth it to fix.  TableSheet is the more descriptive name that is preferred for new usage.  But Sheet is fine to match, as you say.